feat: CchSearchTextField 컴포저블 추가

lluke0 opened the pull request:
## 📌 PR 요약

🌱 작업한 내용
- `CchSearchTextField` 컴포저블을 추가하여 검색 기능을 위한 텍스트 필드를 구현했습니다.
- 입력 상태(Before, Typing, After)에 따라 다른 스타일을 적용합니다.
- `CchSearchTextFieldDefaults`를 통해 기본 색상과 모양을 정의합니다.
- 관련 미리보기(`CchSearchTextFieldPreview`)를 추가했습니다.
- 검색 아이콘(`ic_search.xml`)을 수정했습니다.

🌱 PR 포인트
-

## 📸 스크린샷
|스크린샷|
|:--:|
|<img width="1080" height="736" alt="image" src="https://github.com/user-attachments/assets/eb83d669-4f9e-451e-aa46-4bc78f68b5b1" />|

## 📮 관련 이슈
- Resolved: #8


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **신규 기능**
  * Jetpack Compose UI에서 사용할 수 있는 새로운 검색 텍스트 필드 컴포넌트가 추가되었습니다. 입력 전, 입력 중, 입력 완료의 세 가지 상태를 지원하며, 각 상태에 따라 시각적으로 변화합니다.
  * 검색 텍스트 필드의 다양한 상태를 미리 볼 수 있는 프리뷰 화면이 추가되었습니다.

* **스타일**
  * 검색 아이콘의 디자인과 색상이 변경되었습니다.

* **문서**
  * 마크다운 문서의 코드 블록 표기가 올바르게 수정되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

lluke0 committed:
feat: CchSearchTextField 컴포저블 추가

- `CchSearchTextField` 컴포저블을 추가하여 검색 기능을 위한 텍스트 필드를 구현했습니다.
- 입력 상태(Before, Typing, After)에 따라 다른 스타일을 적용합니다.
- `CchSearchTextFieldDefaults`를 통해 기본 색상과 모양을 정의합니다.
- 관련 미리보기(`CchSearchTextFieldPreview`)를 추가했습니다.
- 검색 아이콘(`ic_search.xml`)을 수정했습니다.

BEEEAM-J reviewed (commented):
"ChukChukRegularTextField" 컴포넌트에서 우측 아이콘만 변경해도 괜찮을 거 같은데, 검색용 텍스트 필드를 따로 추가하신 이유가 있을까요?

lluke0 commented:
> "ChukChukRegularTextField" 컴포넌트에서 우측 아이콘만 변경해도 괜찮을 거 같은데, 검색용 텍스트 필드를 따로 추가하신 이유가 있을까요?

`ChukChukRegularTextField`가 수정되면 `SearchTextField`도 영향을 받기 때문에 분리하였습니다. 피그마 상에서도 서로 분리가 되어있기 때문에 분리하는게 맞을 것 같아요! 또 디자인에서 horizontalPadding 값이 서로 다르더라구용

lluke0 committed:
feat: 검색 TextField UI 개선

- `CchSearchTextField`의 검색 아이콘과 텍스트 필드 사이에 간격 추가
- 미리보기 `TextField.kt`의 `afterValue` 텍스트 길이 증가

BEEEAM-J commented:
> > "ChukChukRegularTextField" 컴포넌트에서 우측 아이콘만 변경해도 괜찮을 거 같은데, 검색용 텍스트 필드를 따로 추가하신 이유가 있을까요?
> 
> `ChukChukRegularTextField`가 수정되면 `SearchTextField`도 영향을 받기 때문에 분리하였습니다. 피그마 상에서도 서로 분리가 되어있기 때문에 분리하는게 맞을 것 같아요! 또 디자인에서 horizontalPadding 값이 서로 다르더라구용

아하 세부적인 디자인은 제가 확인을 안한 상태였어서 하나로 사용해도 괜찮을거라고 생각했었습니다!
확인 감사합니다 :)

BEEEAM-J reviewed (approved):
고생하셨습니다~!
